### PR TITLE
pcsx2: multiple improvements

### PR DESCRIPTION
**Pull request by samuelgrf**
###### Motivation for this change
Build pcsx2 with `gtk3`, this is already done on both [Arch](https://github.com/archlinux/svntogit-community/blob/9a92a0c12e437cbfa4ad6626c3d27fb248658590/trunk/PKGBUILD#L72) and [Gentoo](https://gitweb.gentoo.org/repo/gentoo.git/tree/games-emulation/pcsx2/pcsx2-1.6.0.ebuild#n82).
I updated the pcsx2 version to master, because it would otherwise crash with `trace trap (core dumped)` when opening menus.

I also enabled x86_64 builds, since it is supported on master.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samuelgrf** [comment]
The last commit enables building with LTO, this should be tested on darwin.

File: pkgs/misc/emulators/pcsx2/default.nix [hunk 1]
```diff
@@ -1,53 +1,56 @@
...
+    "-DwxWidgets_LIBRARIES=${wxGTK}/lib"
     "-DXDG_STD=TRUE"
-    "-DGTK2_GLIBCONFIG_INCLUDE_DIR=${glib.out}/lib/glib-2.0/include"
-    "-DGTK2_GDKCONFIG_INCLUDE_DIR=${gtk2.out}/lib/gtk-2.0/include"
-    "-DGTK2_INCLUDE_DIRS=${gtk2.dev}/include/gtk-2.0"
-    "-DGTK3_API=FALSE"
   ];
 
+  patchPhase = ''
+    substituteInPlace cmake/BuildParameters.cmake \
```
Comment (jtojnar) [inline pkgs/misc/emulators/pcsx2/default.nix:36]: Overriding `patchPhase` breaks `patches` field. Use `prePatch` or `postPatch` instead.

**Comment by hrdinka** [comment]
Thanks for the changes and sorry for not responding earlier. It somehow got lost in my mails and I completely missed this PR till today. I am currently on the road, but will give it a try and thorough look soon (hopefully tomorrow).

**Comment by hrdinka** [comment]
Tested locally. One less GTK2 App and a lot of gone problems, thanks for all the changes! :+1:

**Pull request merged**